src/smart-components/group/add-group/roles-list.js

```diff
@@ -26,11 +25,15 @@ const createRows = (data, expanded, checkedRows = []) => {
     : [];
 };
 
-const RolesList = ({ roles, fetchRoles, isLoading, pagination, selectedRoles, setSelectedRoles }) => {
+const RolesList = ({ roles, fetchRoles, isLoading, pagination, selectedRoles, canSort, setSelectedRoles }) => {
   const [filterValue, setFilterValue] = useState('');
+  const { current: columns } = useRef([
```

Comment:
I think you don't want to recalculate these rows on each render right? Wouldn't it be better to use `useCallback`?
```JS
const columns = useCallback(() => ([
  { title: 'Name', key: 'display_name', ...(canSort ? { transforms: [sortable] } : { orderBy: 'name' }) },
  { title: 'Description' },
]),[ canSort ])();

Reply:
Yup, that is why I am using the ref. It's static and you don't need any function calls.